shared/oae/api/oae.api.js

```diff
@@ -150,19 +150,31 @@ define(['oae.api.admin', 'oae.api.authentication', 'oae.api.config', 'oae.api.co
         };
 
 
-        //////////////////////////
-        // Terms and Conditions //
-        //////////////////////////
+        /////////////////////////////////////////
+        // User details & Terms and Conditions //
+        /////////////////////////////////////////
 
         /**
-         * Trigger the Terms and Conditions widget if the Terms and Conditions
-         * need to be accepted before using the system.
+         * Trigger the user details widgets if the user needs to provide additional profile information
+         * or has to accept the Terms and Conditions
          */
-        var setUpTermsAndConditions = function() {
-            if (oae.data.me.needsToAcceptTC) {
-                // Insert the terms and conditions widget in settings mode
+        var setupUserDetails = function() {
+            if (oae.data.me.anon) {
+                return;
+            }
+
+            var needsToProvideDisplayName = (!oae.data.me.displayName || oae.data.me.displayName.indexOf('http://') !== -1);
```

Comment:
The valid display name check should be centralized into some utility. E.g.,

`oae.api.util.isValidDisplayName(oae.data.me.displayName);`